include/sys/__assert.h

```diff
@@ -20,12 +20,38 @@
 #define __ASSERT_ON 0
 #endif
 
+#if defined(CONFIG_ASSERT_VERBOSE)
+#define __ASSERT_PRINT(fmt, ...) printk(fmt, ##__VA_ARGS__)
+#else /* CONFIG_ASSERT_VERBOSE */
+#define __ASSERT_PRINT(fmt, ...)
+#endif /* CONFIG_ASSERT_VERBOSE */
+
 #ifdef CONFIG_ASSERT_NO_FILE_INFO
 #define __ASSERT_FILE_INFO ""
+#define __ASSERT_LINE_INFO 0
 #else  /* CONFIG_ASSERT_NO_FILE_INFO */
 #define __ASSERT_FILE_INFO __FILE__
+#define __ASSERT_LINE_INFO __LINE__
```

Comment:
It would be cool (or another PR) to make this slightly finer grained: That we can indepdently change if we want FILE or LINE. FILE is overkill, but LINE can be helpful when the compiler is a bit too intelligent.

Reply:
Do you think that is needed? Lines add a lot of code-space, more than file info (I think, haven't checked yet). So when you want file info you might as well have both. How do you intend to only use line?  or only file?

Reply:
Correct, but lines will typically be embedded as immediate values and thus occupy PMEM/.text, where FILE will most likely be DMEM/.rodata.
But Oticon may be the only customer for this. No need to fix.